pkg/daemon/daemon.go

```diff
@@ -2807,3 +2810,80 @@ func cipherOrder() []uint16 {
 
 	return append(first, second...)
 }
+
+type rpmOstreeStatus struct {
+	Deployments []struct {
+		RequestedPackages []string `json:"requested-local-packages"`
+	} `json:"deployments"`
+}
+
+func (dn *Daemon) getUnsupportedPackages() {
+	cmd := exec.Command("rpm-ostree", "status", "--json")
+	output, err := cmd.Output()
+	if err != nil {
+		klog.Errorf("Failed to run rpm-ostree status: %v", err)
+		return
+	}
+
+	var status rpmOstreeStatus
+	if err := json.Unmarshal(output, &status); err != nil {
+		klog.Errorf("Failed to parse rpm-ostree status output: %v", err)
+		return
+	}
+
+	// List of supported packages
+	supportedPackages := map[string]bool{
+		"usbguard":             true,
+		"sandboxed-containers": true,
+		"kerberos":             true,
+		"ipsec":                true,
+		"wasm":                 true,
+		"sysstat":              true,
+	}
+
+	if len(status.Deployments) > 0 {
+		requestedLocalPackages := status.Deployments[0].RequestedPackages
+		klog.Infof("Found %d requested local packages in the first deployment", len(requestedLocalPackages))
+
+		// Reset the unsupportedPackagesByVendor metric to start fresh
+		unsupportedPackages.Reset()
+
+		for _, pkg := range requestedLocalPackages {
+			// Check if the package is in the supported list
+			if supportedPackages[pkg] {
+				klog.Infof("Skipping supported package: %s", pkg)
+				continue
+			}
+
+			// Find vendor for unsupported packages
+			vendor, err := dn.getPackageVendor(pkg)
+			if err != nil {
+				klog.Errorf("Failed to get vendor for package %s: %v", pkg, err)
+				continue
```

Comment:
I think if we don't find the vendor, we can probably still add it to the list of packages (i.e. just set vendor to empty but don't `continue` here, just so we can add that to the metrics regardless. WDYT?

Reply:
yeah i agree- good catch